pkg/cmd/cli/cmd/deploy.go

```diff
@@ -325,60 +321,14 @@ func (o DeployOptions) retry(config *deployapi.DeploymentConfig) error {
 
 // cancel cancels any deployment process in progress for config.
 func (o DeployOptions) cancel(config *deployapi.DeploymentConfig) error {
-	if config.Spec.Paused {
-		return fmt.Errorf("cannot cancel a paused deployment config")
-	}
-	deployments, err := o.kubeClient.ReplicationControllers(config.Namespace).List(kapi.ListOptions{LabelSelector: deployutil.ConfigSelector(config.Name)})
-	if err != nil {
-		return err
-	}
-	if len(deployments.Items) == 0 {
-		fmt.Fprintf(o.out, "There have been no deployments for %s/%s\n", config.Namespace, config.Name)
-		return nil
-	}
-	sort.Sort(deployutil.ByLatestVersionDesc(deployments.Items))
-	failedCancellations := []string{}
-	anyCancelled := false
-	for _, deployment := range deployments.Items {
-		status := deployutil.DeploymentStatusFor(&deployment)
-		switch status {
-		case deployapi.DeploymentStatusNew,
-			deployapi.DeploymentStatusPending,
-			deployapi.DeploymentStatusRunning:
-
-			if deployutil.IsDeploymentCancelled(&deployment) {
-				continue
-			}
-
-			deployment.Annotations[deployapi.DeploymentCancelledAnnotation] = deployapi.DeploymentCancelledAnnotationValue
-			deployment.Annotations[deployapi.DeploymentStatusReasonAnnotation] = deployapi.DeploymentCancelledByUser
-			_, err := o.kubeClient.ReplicationControllers(deployment.Namespace).Update(&deployment)
-			if err == nil {
-				fmt.Fprintf(o.out, "Cancelled deployment #%d\n", config.Status.LatestVersion)
-				anyCancelled = true
-			} else {
-				fmt.Fprintf(o.out, "Couldn't cancel deployment #%d (status: %s): %v\n", deployutil.DeploymentVersionFor(&deployment), status, err)
-				failedCancellations = append(failedCancellations, strconv.FormatInt(deployutil.DeploymentVersionFor(&deployment), 10))
-			}
-		}
-	}
-	if len(failedCancellations) > 0 {
-		return fmt.Errorf("couldn't cancel deployment %s", strings.Join(failedCancellations, ", "))
-	}
-	if !anyCancelled {
-		latest := &deployments.Items[0]
-		maybeCancelling := ""
-		if deployutil.IsDeploymentCancelled(latest) && !deployutil.IsTerminatedDeployment(latest) {
-			maybeCancelling = " (cancelling)"
-		}
-		timeAt := strings.ToLower(units.HumanDuration(time.Now().Sub(latest.CreationTimestamp.Time)))
-		fmt.Fprintf(o.out, "No deployments are in progress (latest deployment #%d %s%s %s ago)\n",
-			deployutil.DeploymentVersionFor(latest),
-			strings.ToLower(string(deployutil.DeploymentStatusFor(latest))),
-			maybeCancelling,
-			timeAt)
-	}
-	return nil
+	return rollout.NewRolloutCancel(
```

Comment:
this is pretty ugly, but I don't know how to handle this better (without import loop or accessing the private struct field)